### DISCUSSION TITLE
what version python does this need? is it listed somewhere?

### DISCUSSION THREAD

**Discussion by gsgoldma** [neonbjb/tortoise-tts#265] [General]
i'm getting a huge error during install

(voice) D:\tortoise-tts>python -m pip install -r ./requirements.txt
Collecting tqdm
  Using cached tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
Collecting rotary_embedding_torch
  Downloading rotary_embedding_torch-0.2.1-py3-none-any.whl (4.5 kB)
Collecting transformers==4.19
  Downloading transformers-4.19.0-py3-none-any.whl (4.2 MB)
     ---------------------------------------- 4.2/4.2 MB 16.6 MB/s eta 0:00:00
Collecting tokenizers
  Downloading tokenizers-0.13.2-cp311-cp311-win_amd64.whl (3.3 MB)
     ---------------------------------------- 3.3/3.3 MB 34.9 MB/s eta 0:00:00
Collecting inflect
  Downloading inflect-6.0.2-py3-none-any.whl (34 kB)
Collecting progressbar
  Downloading progressbar-2.5.tar.gz (10 kB)
  Preparing metadata (setup.py) ... done
Collecting einops
  Downloading einops-0.6.0-py3-none-any.whl (41 kB)
     ---------------------------------------- 41.6/41.6 kB 2.0 MB/s eta 0:00:00
Collecting unidecode
  Downloading Unidecode-1.3.6-py3-none-any.whl (235 kB)
     ---------------------------------------- 235.9/235.9 kB ? eta 0:00:00
Collecting scipy
  Downloading scipy-1.10.0-cp311-cp311-win_amd64.whl (42.2 MB)
     ---------------------------------------- 42.2/42.2 MB 8.1 MB/s eta 0:00:00
Collecting librosa
  Downloading librosa-0.9.2-py3-none-any.whl (214 kB)
     ---------------------------------------- 214.3/214.3 kB 6.4 MB/s eta 0:00:00
Collecting numpy==1.20.0
  Downloading numpy-1.20.0.zip (8.0 MB)
     ---------------------------------------- 8.0/8.0 MB 11.0 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [287 lines of output]
      setup.py:66: RuntimeWarning: NumPy 1.20.0 may not yet support Python 3.11.
        warnings.warn(
      Running from numpy source directory.
      setup.py:485: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates
        run_build = parse_setuppy_commands()
      Processing numpy/random\_bounded_integers.pxd.in
      Processing numpy/random\bit_generator.pyx
      Processing numpy/random\mtrand.pyx
      Processing numpy/random\_bounded_integers.pyx.in
      Processing numpy/random\_common.pyx
      Processing numpy/random\_generator.pyx
      Processing numpy/random\_mt19937.pyx
      Processing numpy/random\_pcg64.pyx
      Processing numpy/random\_philox.pyx
      Processing numpy/random\_sfc64.pyx
      Cythonizing sources
      blas_opt_info:
      blas_mkl_info:
      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils
      customize MSVCCompiler
        libraries mkl_rt not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      blis_info:
        libraries blis not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      openblas_info:
        libraries openblas not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
      get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'
      customize GnuFCompiler
      Could not locate executable g77
      Could not locate executable f77
      customize IntelVisualFCompiler
      Could not locate executable ifort
      Could not locate executable ifl
      customize AbsoftFCompiler
      Could not locate executable f90
      customize CompaqVisualFCompiler
      Found executable D:\miniconda3\Library\usr\bin\DF.exe
      customize IntelItaniumVisualFCompiler
      Could not locate executable efl
      customize Gnu95FCompiler
      Could not locate executable gfortran
      Could not locate executable f95
      customize G95FCompiler
      Could not locate executable g95
      customize IntelEM64VisualFCompiler
      customize IntelEM64TFCompiler
      Could not locate executable efort
      Could not locate executable efc
      customize PGroupFlangCompiler
      Could not locate executable flang
      don't know how to compile Fortran code on platform 'nt'
        NOT AVAILABLE

      atlas_3_10_blas_threads_info:
      Setting PTATLAS=ATLAS
        libraries tatlas not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      atlas_3_10_blas_info:
        libraries satlas not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      atlas_blas_threads_info:
      Setting PTATLAS=ATLAS
        libraries ptf77blas,ptcblas,atlas not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      atlas_blas_info:
        libraries f77blas,cblas,atlas not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\system_info.py:1989: UserWarning:
          Optimized (vendor) Blas libraries are not found.
          Falls back to netlib Blas library which has worse performance.
          A better performance should be easily gained by switching
          Blas library.
        if self._calc_info(blas):
      blas_info:
        libraries blas not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\system_info.py:1989: UserWarning:
          Blas (http://www.netlib.org/blas/) libraries not found.
          Directories to search for the libraries can be specified in the
          numpy/distutils/site.cfg file (section [blas]) or by setting
          the BLAS environment variable.
        if self._calc_info(blas):
      blas_src_info:
        NOT AVAILABLE

      C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\system_info.py:1989: UserWarning:
          Blas (http://www.netlib.org/blas/) sources not found.
          Directories to search for the sources can be specified in the
          numpy/distutils/site.cfg file (section [blas_src]) or by setting
          the BLAS_SRC environment variable.
        if self._calc_info(blas):
        NOT AVAILABLE

      non-existing path in 'numpy\\distutils': 'site.cfg'
      lapack_opt_info:
      lapack_mkl_info:
        libraries mkl_rt not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      openblas_lapack_info:
        libraries openblas not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      openblas_clapack_info:
        libraries openblas,lapack not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      flame_info:
        libraries flame not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      atlas_3_10_threads_info:
      Setting PTATLAS=ATLAS
        libraries lapack_atlas not found in D:\miniconda3\envs\voice\lib
        libraries tatlas,tatlas not found in D:\miniconda3\envs\voice\lib
        libraries lapack_atlas not found in C:\
        libraries tatlas,tatlas not found in C:\
        libraries lapack_atlas not found in D:\miniconda3\envs\voice\libs
        libraries tatlas,tatlas not found in D:\miniconda3\envs\voice\libs
        libraries lapack_atlas not found in D:\miniconda3\Library\lib
        libraries tatlas,tatlas not found in D:\miniconda3\Library\lib
      <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
        NOT AVAILABLE

      atlas_3_10_info:
        libraries lapack_atlas not found in D:\miniconda3\envs\voice\lib
        libraries satlas,satlas not found in D:\miniconda3\envs\voice\lib
        libraries lapack_atlas not found in C:\
        libraries satlas,satlas not found in C:\
        libraries lapack_atlas not found in D:\miniconda3\envs\voice\libs
        libraries satlas,satlas not found in D:\miniconda3\envs\voice\libs
        libraries lapack_atlas not found in D:\miniconda3\Library\lib
        libraries satlas,satlas not found in D:\miniconda3\Library\lib
      <class 'numpy.distutils.system_info.atlas_3_10_info'>
        NOT AVAILABLE

      atlas_threads_info:
      Setting PTATLAS=ATLAS
        libraries lapack_atlas not found in D:\miniconda3\envs\voice\lib
        libraries ptf77blas,ptcblas,atlas not found in D:\miniconda3\envs\voice\lib
        libraries lapack_atlas not found in C:\
        libraries ptf77blas,ptcblas,atlas not found in C:\
        libraries lapack_atlas not found in D:\miniconda3\envs\voice\libs
        libraries ptf77blas,ptcblas,atlas not found in D:\miniconda3\envs\voice\libs
        libraries lapack_atlas not found in D:\miniconda3\Library\lib
        libraries ptf77blas,ptcblas,atlas not found in D:\miniconda3\Library\lib
      <class 'numpy.distutils.system_info.atlas_threads_info'>
        NOT AVAILABLE

      atlas_info:
        libraries lapack_atlas not found in D:\miniconda3\envs\voice\lib
        libraries f77blas,cblas,atlas not found in D:\miniconda3\envs\voice\lib
        libraries lapack_atlas not found in C:\
        libraries f77blas,cblas,atlas not found in C:\
        libraries lapack_atlas not found in D:\miniconda3\envs\voice\libs
        libraries f77blas,cblas,atlas not found in D:\miniconda3\envs\voice\libs
        libraries lapack_atlas not found in D:\miniconda3\Library\lib
        libraries f77blas,cblas,atlas not found in D:\miniconda3\Library\lib
      <class 'numpy.distutils.system_info.atlas_info'>
        NOT AVAILABLE

      lapack_info:
        libraries lapack not found in ['D:\\miniconda3\\envs\\voice\\lib', 'C:\\', 'D:\\miniconda3\\envs\\voice\\libs', 'D:\\miniconda3\\Library\\lib']
        NOT AVAILABLE

      C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\system_info.py:1849: UserWarning:
          Lapack (http://www.netlib.org/lapack/) libraries not found.
          Directories to search for the libraries can be specified in the
          numpy/distutils/site.cfg file (section [lapack]) or by setting
          the LAPACK environment variable.
        return getattr(self, '_calc_info_{}'.format(name))()
      lapack_src_info:
        NOT AVAILABLE

      C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\system_info.py:1849: UserWarning:
          Lapack (http://www.netlib.org/lapack/) sources not found.
          Directories to search for the sources can be specified in the
          numpy/distutils/site.cfg file (section [lapack_src]) or by setting
          the LAPACK_SRC environment variable.
        return getattr(self, '_calc_info_{}'.format(name))()
        NOT AVAILABLE

      numpy_linalg_lapack_lite:
        FOUND:
          language = c
          define_macros = [('HAVE_BLAS_ILP64', None), ('BLAS_SYMBOL_SUFFIX', '64_')]

      C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\dist.py:275: UserWarning: Unknown distribution option: 'define_macros'
        warnings.warn(msg)
      running dist_info
      running build_src
      build_src
      building py_modules sources
      creating build
      creating build\src.win-amd64-3.11
      creating build\src.win-amd64-3.11\numpy
      creating build\src.win-amd64-3.11\numpy\distutils
      building library "npymath" sources
      Traceback (most recent call last):
        File "D:\miniconda3\envs\voice\Lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 351, in <module>
          main()
        File "D:\miniconda3\envs\voice\Lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 333, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "D:\miniconda3\envs\voice\Lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 152, in prepare_metadata_for_build_wheel
          return hook(metadata_directory, config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\build_meta.py", line 157, in prepare_metadata_for_build_wheel
          self.run_setup()
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\build_meta.py", line 249, in run_setup
          self).run_setup(setup_script=setup_script)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\build_meta.py", line 142, in run_setup
          exec(compile(code, __file__, 'exec'), locals())
        File "setup.py", line 513, in <module>
          setup_package()
        File "setup.py", line 505, in setup_package
          setup(**metadata)
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\core.py", line 169, in setup
          return old_setup(**new_attr)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\__init__.py", line 165, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 148, in setup
          dist.run_commands()
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 967, in run_commands
          self.run_command(cmd)
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 986, in run_command
          cmd_obj.run()
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\command\dist_info.py", line 31, in run
          egg_info.run()
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\command\egg_info.py", line 24, in run
          self.run_command("build_src")
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 986, in run_command
          cmd_obj.run()
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\command\build_src.py", line 144, in run
          self.build_sources()
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\command\build_src.py", line 155, in build_sources
          self.build_library_sources(*libname_info)
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\command\build_src.py", line 288, in build_library_sources
          sources = self.generate_sources(sources, (lib_name, build_info))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\command\build_src.py", line 378, in generate_sources
          source = func(extension, build_dir)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "numpy\core\setup.py", line 671, in get_mathlib_info
          st = config_cmd.try_link('int main(void) { return 0;}')
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\command\config.py", line 243, in try_link
          self._link(body, headers, include_dirs,
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\command\config.py", line 162, in _link
          return self._wrap_method(old_config._link, lang,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\command\config.py", line 96, in _wrap_method
          ret = mth(*((self,)+args))
                ^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\command\config.py", line 137, in _link
          (src, obj) = self._compile(body, headers, include_dirs, lang)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\command\config.py", line 105, in _compile
          src, obj = self._wrap_method(old_config._compile, lang,
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\command\config.py", line 96, in _wrap_method
          ret = mth(*((self,)+args))
                ^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\command\config.py", line 132, in _compile
          self.compiler.compile([src], include_dirs=include_dirs)
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\_msvccompiler.py", line 401, in compile
          self.spawn(args)
        File "C:\Users\Gregory\AppData\Local\Temp\pip-build-env-eyz_y93i\overlay\Lib\site-packages\setuptools\_distutils\_msvccompiler.py", line 505, in spawn
          return super().spawn(cmd, env=env)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Gregory\AppData\Local\Temp\pip-install-1o7_3bm2\numpy_33f33b2f32ec440d85daa961368f6961\numpy\distutils\ccompiler.py", line 90, in <lambda>
          m = lambda self, *args, **kw: func(self, *args, **kw)
                                        ^^^^^^^^^^^^^^^^^^^^^^^
      TypeError: CCompiler_spawn() got an unexpected keyword argument 'env'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

(voice) D:\tortoise-tts>
(voice) D:\tortoise-tts>
(voice) D:\tortoise-tts>conda install pytorch torchvision torchaudio pytorch-cuda=11.6 -c pytorch -c nvidia
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: -
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
Examining pytorch-cuda=11.6:  25%|█████████████▊                                         | 2/8 [00:13<00:41,  6.91s/it]-failed

UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

**Comment by cyberofficial** [comment]
I was able to get it running under Python 3.8.16 for windows

**Reply by gsgoldma** [reply]
thanks that worked!
